nt-web-app/identity/identity.ts

```diff
@@ -0,0 +1,28 @@
+import { getUser, getUsersById, getRedirectURL, handleRedirectResponse, validateAuthServer, provider } from './twitch';
```

Comment:
I believe you asked a question somewhere about how to generalize this so that it doesn't require a code change to switch providers. Here's one way you can do that.

1) define each identity provider in a map (in this case, I'll use an object for convenience)
2) create a function to return the requested provider by its name

Example:

(current code)
```ts
import * as twitch from './twitch';
import * as google from './google';

const providers = {twitch, google};

export const getIdentityProvider = (name: keyof typeof providers) => providers[name];

export const getConfiguredProvider = () => {
  const name = process.env.IDENTITY_PROVIDER ?? 'twitch';
  if (Object.prototype.hasOwnProperty.call(providers, name)) return getIdentityProvider(name as any);
  throw new Error(`No such identity provider: '${name}'. Expected: ${Object.keys(providers).join(', ')}`);
}
```

`getConfiguredProvider()` will resolve to the set of exported functions using an environment variable to select the provider to use, or fail. `getIdentityProvider('name')` will return a specific provider.

If you want to ensure better conformance / compatibility using typescript, you can define the providers slightly differently to conform to an interface:

`identity.ts`:

```ts
import { twitchProvider } from './twitch';
import { googleProvider } from './google';

const providers = {
  twitch: twitchProvider as any as IdentityProvider<'twitch', UserData>,
  google: googleProvider
} satisfies Record<string, IdentityProvider<string, UserData>>;

export const getIdentityProvider = <T extends keyof typeof providers>(name: T): typeof providers[T] => providers[name];

export const getConfiguredProvider = (): typeof providers[keyof typeof providers] => {
  const name = process.env.IDENTITY_PROVIDER ?? 'twitch';
  if (Object.prototype.hasOwnProperty.call(providers, name)) return getIdentityProvider(name as any);
  throw new Error(`No such identity provider: '${name}'. Expected: ${Object.keys(providers).join(', ')}`);
}

export interface UserData {
  //Subscribder Id
  sub: string,
  preferred_username: string,
  picture?: string,
  email?: string,
}

export interface IdentityProvider<Name extends string = string, T extends UserData = UserData> {
  getUser(token: string): Promise<T>;
  getUsersById(userIds: string[]): Promise<T[]>;
  getRedirectURL(redirectURL: string, scope: string, state: string): string;
  handleRedirectResponse(code: string, state: string, redirectUri: string, grantType: string): Promise<T>;
  validateAuthServer(): Promise<Boolean>;
  provider: Name;
}
```

`google.ts`:
```ts
import { type IdentityProvider, type UserData } from "./identity";

export const googleProvider: IdentityProvider<'google', UserData & {'googleSpecificStuff': 'etc'}> = {
  getUser,
  getUsersById,
  getRedirectURL,
  handleRedirectResponse,
  validateAuthServer,

  provider,
}
```

Doing it this way will ensure that he functions exported by the identity provider conform to the defined interface in such a way that Typescript won't build if you've made an error. However, it also allows for specialization on top of that, as well as type discrimination:

<img width="839" alt="image" src="https://github.com/Noita-Together/noita-together/assets/1588419/c5cd8fc7-3467-4836-a941-aa59f907df4e">


However, it also allows for specialization: if the Google provider